specs/commandline/sign.md

```diff
@@ -4,7 +4,21 @@
 
 Use `notation sign` to sign artifacts.
 
-Signs an OCI artifact that is stored in a registry. Upon successful signing, the generated signature is pushed to the registry with the digest of the OCI artifact returned.
+Signs an OCI artifact that is stored in a registry. Always use a `digest` to identify an artifact. `Tags` are mutable, but `digests` uniquely and immutably identify artifacts. If a tag is used, notation resolves the tag to the `digest` before signing.
+
+Upon successful signing, the generated signature is pushed to the registry and associated with the signed OCI artifact. The output message is printed out as following:
+
+```text
+Sign succeeded. Signature has been attached to <registry>/<repository>@<digest>.
+```
+
+If a `tag` is used to identify the OCI artifact, the output message is as following:
+
+```test
+Warning: Tag is used. Always use digest to identify the reference uniquely and immutably.
```

Comment:
I think we can improve on the error message. I would recommend following some guidelines (here a good example https://spark.apache.org/error-message-guidelines.html). I would recommend improving this as follows:

```
Warning: A tag is used to identify the artifact for signing. Artifact tags are mutable. Use digests to uniquely identify artifacts and avoid mutability.
Resolving artifact tag <tag> to digest <digest> before signing.
Signing of artifact <artifact-as-provided-in-the-command> succeeded and the signature is attached to <registry>/<repository>@<digest>.
```

Also, this seems to be specific to the image signing. What about other artifacts?

Reply:
Thanks Toddy for sharing the guidelines. It is very useful. (Finally, I got the guidelines, I was searching for a while)

I have questions on `<artifact-as-provided-in-the-command>`, once tag is resolved to digest, should we use the same provided in the command or just replacing the tag with digest?

Reply:
Currently we only support signing remote artifacts, so the artifacts are identified by <registry>/<repository>@<digest>. It is the same signing logical as container images. Is there anything you see from output message indicates the artifact is a container image?

Reply:
Regarding the first comment, the reason I typed `<artifact-as-provided-in-the-command>` is because the user can type the tag for the artifact in the command. Here are the steps:
```
notation sign test.mycr.io/myimage:tag
```
The message should look like:
```
Warning: A tag is used to identify the artifact for signing. Artifact tags are mutable. Use digests to uniquely identify artifacts and avoid mutability.
Resolving artifact tag 'tag' to digest 'sha:1234567890` before signing.
Signing of artifact test.mycr.io/myimage:tag succeeded and the signature is attached to test.mycr.io/myimage@sha1234567890.
```
This way, we give the user information what they typed and what changes were made as part of the command. Hope this helps.